server/src/services/trainingLinks.service.ts

```diff
@@ -25,123 +29,169 @@ interface ILinks {
 }
 
 const lbaDomain = "https://labonnealternance.apprentissage.beta.gouv.fr"
+const lbaCandidat = new URL(`${lbaDomain}/recherche-emploi`)
```

Comment:
ca fonctionne actuellement mais c'est pas thread safe. Ca peut amener des bugs de race condition si on utilise des promises plus tard.
```suggestion
const lbaCandidat = () => new URL(`${lbaDomain}/recherche-emploi`)
```

Reply:
il faut modifier les caller en conséquence

Reply:
encore mieux :
```suggestion
const buildEmploiUrl = (params: Record<string, string>) => {
  const url = new URL(`${lbaDomain}/recherche-emploi`)
  Object.entries(params).forEach(([key, value]) => url.searchParams.append(key, value))
  return url.toString()
}
```